### DISCUSSION TITLE
Netconn/conn received data ready

### DISCUSSION THREAD

**Discussion by davideferrario** [MaJerle/lwesp#127] [General]
Is it possible to add a new function to conn/netconn API to get the amount of data received on a connection?

Just to be clear: to get the answer of AT+CIPRECVLEN? command.

**Comment by MaJerle** [comment]
What is the good reason for this? Today library automatically reads data from ESP, if there is possibility to allocate memory for packet buffer.

**Reply by davideferrario** [reply]
For example to know if a complete or partial answer from a command has been received.

**Reply by MaJerle** [reply]
Can you propose a real use case? I have troubles to understand, since new packets can arrive from remote to the ESP while read command is in progress, for instance.
